patient_data_source.py

```diff
@@ -0,0 +1,214 @@
+		return self.UNIT_CODES[self.get_observation_type()]
+
+	def get_loinc_code(self) -> str:
+		"""Returns a computer processable form for the ObservationType. Default implementation uses the LOINC_CODES."""
+		return self.LOINC_CODES[self.get_observation_type()]
+
+	@abstractmethod
+	def get_value(self) -> str:
+		"""Returns the Observation's value."""
+		pass
+
+	def get_time(self) -> datetime:
+		"""Returns the observation's recorded time."""
+		return datetime.min
+
+class PatientDataSource(ABC):
+	"""Abstract class for loading patient data into a smart FHIR server"""
+	
+	@abstractmethod
+	def get_all_patients(self) -> Iterable[Patient]:
+		"""Returns a list of all Patients."""
+		pass
+
+	@abstractmethod
+	def get_patient_observations(self, patient : Patient) -> Iterable[Observation]:
+		"""Returns a list of all Observation for one patient."""
+		pass
+
+	def create_patient(self, patient: Patient) -> FHIR_Patient :
+		"""Create a smart FHIR_Patient object."""
+		gender = patient.get_gender().name
+		family, given = patient.get_name()
+		date = FHIRDate(patient.get_dob().isoformat())
+
+		fhir_patient_args = {
+		  'gender' : gender,
+		  'birthDate' : date.isostring,
+		  'name' : [{'family':family,'given':[given]}],
+		}
+
+		identifier_system = patient.get_identifier_system()
+		identifier_value = patient.get_identifier_value()
+
+		if (identifier_system is not None and identifier_value is not None):
+			fhir_patient_args['identifier'] = [{
+				'system': identifier_system,
+				'value': identifier_value
+			}]
+		elif (identifier_system is not None):
+			fhir_patient_args['identifier'] = [{
+				'system': identifier_system
+			}]
+		elif (identifier_value is not None):
+			fhir_patient_args['identifier'] = [{
+				'value': identifier_value
+			}]
+
+		return FHIR_Patient(fhir_patient_args)
+
+	def create_observation(self, observation : Observation, patient_id : str) -> FHIR_Observation : 
+		"""Create a smart FHIR_Observation object. patient_id is an ID value of a Patient item currently on the FHIR server"""
+		unique_id = observation.get_identifier_value()
+		value = observation.get_value()
+		unit_string = observation.get_unit_string()
+		unit_code = observation.get_unit_code()
+		loinc = observation.get_loinc_code()
+		display_string = observation.get_display_string()
+		system = observation.get_identifier_system()
+		date = FHIRDate(observation.get_time().isoformat())
+
+		fhir_observation_args = {
+		  'code' : {
+		    'coding' : [
+		      {'code': loinc, 'display': display_string, 'system': 'http://loinc.org'}
+		    ]
+		  },
+		  'status' :'final',
+		  'subject': {'reference': f'Patient/{patient_id}'},
+		  'valueQuantity': {
+		    'code': unit_code,
+		    'system': 'http://unitsofmeasure.org',
+		    'unit': unit_string,
+		    'value': value
+		  },
+		  'effectiveDateTime': date.isostring,
+		}
+
+		identifier_system = observation.get_identifier_system()
+		identifier_value = observation.get_identifier_value()
```

Comment:
these two were already defined above as `unique_id` and `system`. I would delete redundant ones above